### ISSUE TITLE
Add symmetric set difference operator

### ISSUE THREAD

**Issue by will62794** [tlaplus/CommunityModules#46]
The symmetric difference of two sets can be a convenient operation to have around. `FiniteSetsExt.tla` seems a good place to add it. It can be defined as follows:

```tla
SymmDiff(A, B) == (A \ B) \cup (B \ A)
```

**Comment by lemmy** [comment]
Make sense to me!  You should have received an invite to become a maintainer of this repository with which you can make the proposed changes. ;-)

**Comment by will62794** [comment]
Awesome, thanks! I'll push a commit soon.

**Comment by johnyf** [comment]
Thank you for the definition. I would like to suggest considering naming the operator `SymDiff`.

**Comment by johnyf** [comment]
There exists also a tool called `symdiff`: https://symdiff.github.io

**Comment by muenchnerkindl** [comment]
Thanks for the suggestion. One minor gripe I have is that the definition can be useful beyond finite sets, so FiniteSetsExt may not be the most intuitive place (but we don't have another module for set theoretic operators, and the same can be said about the existing ChooseUnique).

**Comment by lemmy** [comment]
I don't have a strong opinion about naming, but a fair number of the operators in  CommunityModules are not abbreviated.  Long operators names also make it less likely to cause name clashes.

**Comment by johnyf** [comment]
I think `SymmetricDifference` would be good. Searching for "sym" and "diff" through the `CommunityModules` repository, `Examples` repository, `tlapm/examples*` directories,  and `tlapm/library` directory, it seems that the full words are used everywhere (identifiers that contain `Symmetric` and `Symmetry`, together with other words; symmetry appears more frequently in identifiers than difference).

Modules that extend the module where `SymmetricDifference` is defined can define an abbreviation for local use, e.g., `SymDiff(A, B) == SymmetricDifference(A, B)`. Perhaps a suggested abbreviation could be mentioned in a comment adjacent to the definition of the operator `SymmetricDifference`.

**Comment by johnyf** [comment]
Thinking of which word might express the symmetric difference, `Nonoverlap` seems to be one. This capitalization is as [one word](https://en.wiktionary.org/wiki/nonoverlap), an alternative is `NonOverlap`, though the latter may be read as alluding to a predicate, i.e., "non-overlapping", whereas the former suggests a noun.

"Symmetric difference" appears to be widely used terminology, so I am not sure whether "nonoverlap" would be as easily searched-for in the files. "Symmetric difference" could be mentioned in case the word "nonoverlap" is used, both to describe the operator's meaning, and to facilitate searching.

There are at least two papers that mention "non[-]overlap" as "set difference", quoting from each:

- In https://doi.org/10.1175/WAF-D-13-00135.1 the following appears:
  > The symmetric difference is the total nonoverlap area between the matched objects,

- In https://doi.org/10.1049/iet-cvi.2017.0277 and https://arxiv.org/pdf/1510.04563.pdf the following appears:
  > the area of mutual non-overlap, i.e. the area of the symmetric difference (aosd) of the compared shapes.

In Python, the builtin `set` type has a (unary) method called [`symmetric_difference`](https://docs.python.org/3/library/stdtypes.html#frozenset.symmetric_difference).

**Comment by will62794** [comment]
Ok, so it sounds like we are in support of adding the following definition
```tla
SymDiff(A, B) == (A \ B) \cup (B \ A)
```
to `FiniteSetsExt`. As @muenchnerkindl pointed out, eventually it might be nice to have a clearer separation between finite set operators and more general set operators, but until we do that it seems fine to place it in `FiniteSetsExt`.

**Comment by will62794** [comment]
Addressed by 8d661bc.
